quinn-proto/src/connection/streams.rs

```diff
@@ -523,8 +534,12 @@ impl Streams {
             }
 
             let meta = frame::StreamMeta { id, offsets, fin };
-            trace!(id = %meta.id, off = meta.offsets.start, len = meta.offsets.end - meta.offsets.start, fin = meta.fin, "STREAM");
-            meta.encode(true, buf);
+            let len = (meta.offsets.end - meta.offsets.start) as usize;
+            trace!(id = %meta.id, off = meta.offsets.start, len, fin = meta.fin, "STREAM");
+            // This is the last frame, and can therefore omit the length tag, if there are no
+            // following stream frames and no padding will be needed.
+            let is_last_frame = self.pending.is_empty() && buf.len() + len >= min_buf_size;
```

Comment:
Is this intentionally unnecessarily restrictive? For example, if `self.pending.is_empty()` is `false` but we're filling up to within `frame::Stream::SIZE_BOUND` of the `max_buf_size`, we could also say `is_last_frame`.

Reply:
No, good catch. We should have a similar fix for the datagram case as well.

Reply:
Applied.